app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
+
+  after_create :update_user_posts_counter
+
+  def most_recent_comments
+    comments.order(created_at: :desc).order(5)
+  end
```

Comment:
- [ ] This method was not functioning as expected, all the posts were listed when I tried to check your work via `rails c` command. There is a mistake in the syntax. If you want to order `comments` by the `created_at` attribute in descending order and limit the result to 5 comments, you might use the following syntax:

```suggestion

  def most_recent_comments
    comments.order(created_at: :desc).limit(5)
  end
```